src/main/java/com/griddynamics/serialDeserialTask/SerialDeserial.java

```diff
@@ -0,0 +1,242 @@
+            JsonObject vehicleObject = json.getAsJsonObject();
+            JsonElement vehicleTypeElement = vehicleObject.get(vehicleTypeElementName);
+
+            Class<? extends Vehicle> vehicleType = vehicleTypeRegistry.get(vehicleTypeElement.getAsString());
+            return gson.fromJson(vehicleObject, vehicleType);
+        }
+    }
+
+    public static void readFromCreatedFile(URL resource) throws IOException {
+        String line = Files.readAllLines(Paths.get(resource.getPath()).getParent().resolve("new_format_vehicles.json")).get(0);
+        VehicleDeserializer deserializer = new VehicleDeserializer("typeOfVehicle");
+        deserializer.registerGarageType("CAR", Car.class);
+        deserializer.registerGarageType("TRUCK", Truck.class);
+        Gson gson = new GsonBuilder().registerTypeAdapter(Vehicle.class, deserializer).create();
+        List<Vehicle> outList = gson.fromJson(line, new TypeToken<List<Vehicle>>() {
+        }.getType());
+        for (Vehicle cur : outList) {
+            System.out.println(cur);
+        }
+    }
+
+    public static void writeIntoOneFile(List<Vehicle> vehicles, URL resource) {
+        try (Writer vehiclesWriter = new FileWriter(Paths.get(resource.getPath())
+                .getParent().resolve("new_format_vehicles.json").toFile())) {
+            Gson gson = new Gson();
+            String jsonStr = gson.toJson(vehicles);
+            vehiclesWriter.write(jsonStr);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public static List<Vehicle> readFromMainFile(String inputFile) throws FileFormatException, IOException {
+        Path inputFilePath = Paths.get(inputFile);
+        List<String> lines = Files.readAllLines(inputFilePath);
+        List<Vehicle> vehicles = new ArrayList<>();
+        for (int i = 0; i < lines.size(); i++) {
+            String line = lines.get(i);
+            String cur = line.replaceAll("\\s+", "");
+            cur = cur.replace(':', ',');
+            String[] parameters = cur.split(",");
+            if (parameters.length == 5) {
+                if (parameters[0].equalsIgnoreCase("CAR")) {
+                    vehicles.add(new Car(parameters[1], parameters[2],
+                            Integer.parseInt(parameters[3]), parameters[4]));
+                } else {
+                    vehicles.add(new Truck(parameters[1], parameters[2],
+                            Integer.parseInt(parameters[3]), Long.parseLong(parameters[4])));
+                }
+            } else {
+                throw new FileFormatException("Your values in file are invalid. " +
+                        "Check them and try again.", i + 1);
+            }
+        }
+        if (vehicles.isEmpty()) {
+            System.out.println("Empty file provided.");
+        }
+        return vehicles;
+    }
+
+
+    public interface MovingObject {
+        boolean isMoving();
+    }
+
+    public static class Vehicle implements MovingObject {
```

Comment:
@kolegova-mary also, let's move all classes and interface outside from SerialDeserial class, it becomes difficult to read the code